tensorflow/core/graph/mkl_graph_util.h

```diff
@@ -139,7 +159,7 @@ static inline bool IsMklLayoutDependentOp(const string& op_name, DataType T) {
 #ifdef ENABLE_INTEL_MKL_BFLOAT16
   // Restrict regular ops to FLOAT and BFLOAT16
   if (kernel.find(kMklLayoutDependentOpLabelPattern) != string::npos) {
-    return (T == DT_FLOAT || T == DT_BFLOAT16);
+    return (T == DT_FLOAT || (T == DT_BFLOAT16 && HasBfloat16Support(T)));
```

Comment:
Sorry for misunderstanding the purpose of the function. If the function is checking the type anyway, then I think `CheckBfloat16Support` is a better name. We should also remove `T == DT_BFLOAT16` part. Same for the other check in this file. Sorry for the inconvenience!
 ```suggestion
    return (T == DT_FLOAT || CheckBfloat16Support(T));
```